More tests

povilasb opened the pull request:
I started out with inspecting changes needed to solve https://github.com/maidsafe/crust/issues/889
Then before doing first changes I decided to extract the code that will be modified and have some tests for it.
Also added some refactoring changes.

povilasb committed:
refactor/listener: extract some code

Makes it more testable and extensible.

povilasb committed:
refactor/tests: extract some code

maidsafe-highfive commented:
Thanks for the pull request, and welcome! The MaidSafe team is excited to review your changes, and you should hear from @canndrew (or someone else) soon.

If any changes to this PR are deemed necessary, please add them as extra commits. This ensures that the reviewer can see what has changed since they last reviewed the code. The way GitHub handles out-of-date commits, this should also make it reasonably obvious what issues have or haven't been addressed. Large or tricky changes may require several passes of review and changes.

Please see [CONTRIBUTOR.md](../blob/master/CONTRIBUTOR) for more information.

povilasb committed:
chore/deps: reformat Cargo.toml with 'cargo edit'

povilasb committed:
refactor/lints: update maplit

New maplit has unused_results fixed.

povilasb committed:
chore/deps: remove hardcoded dependencies

These versions were required with earlier Rust versions. But since we updated Rust, these version specializations are obsolete.

povilasb committed:
chore/deps: use tilde version requiremenet

ustulation commented:
OS X is hung hence cancelled - passes on the rest.
